File: tcp-udp/tcp/labirinto.c
#include <stdio.h>



//parametros:   labirinto,             qtd linhas, qtd colunas, pos Atual X e Y
int labirinto_bt(int labirinto[20][20], int linha, int coluna, int x, int y){
    if(x==-1 || y == -1 || x == linha || y == coluna || labirinto[y][x] != 0) return 0;

    int ans = labirinto_bt(labirinto, linha, coluna, x, y+1);
    if(ans == 0) labirinto_bt(labirinto, linha, coluna, x + 1, y);
    
    return 1;
}

int main(){
    int linha, coluna, labirinto[20][20];
    scanf("%d %d", &linha, &coluna);
    for (int i = 0; i < linha; ++i)
        for (int j = 0; j < coluna; ++j)
            scanf("%d", &labirinto[i][j]);
    int saida = labirinto_bt(labirinto, linha, coluna, 0, 0);
    printf("%d\n", saida);

    return 0;
}

File: tcp-udp/tcp/labirinto.py
"0 0 0 0 0 0 1 0 1 0 1 1 0 0 1 0 0 0"
labirinto = [
    [10,10,0,0,0,0],
    [1,10,1,0,1,1],
    [10,10,1,0,0,0]
]

# x,y --> x: posicaoLinha, y: posicaoColuna
def labirinto_bt(labirinto, qtd_linha, qtd_coluna, x, y):
    if(x == qtd_linha or y == qtd_coluna or labirinto[x][y] == 1 or x == -1 or y == -1): return 0
    # desce
    if(x == qtd_linha  and y == qtd_coluna): return 1

    labirinto[x][y] = 10
    ans = labirinto_bt(labirinto, qtd_linha, qtd_coluna, x-1, y)
    if(ans == 0): 
        ans = labirinto_bt(labirinto, qtd_linha, qtd_coluna, x+1, y)
    if(ans == 0): 
        ans = labirinto_bt(labirinto, qtd_linha, qtd_coluna, x, y-1)
    if(ans == 0): 
        ans = labirinto_bt(labirinto, qtd_linha, qtd_coluna, x, y+1)
    labirinto[x][y] = 0

    return ans

tem_saida = labirinto_bt(labirinto,3,6,0,0)
print(tem_saida)